### PR TITLE
Run standard (sharable) REST Catalog integrations tests with MinIO

### PR DESCRIPTION
**Pull request by dimas-b**
Keep special edge case tests (path style, STS endpoint and client-side FileIO) in `QuarkusRestCatalogMinIoSpecialIT`.

Contributes to #1530

### CODE REVIEW

File: integration-tests/src/main/java/org/apache/polaris/service/it/test/PolarisRestCatalogIntegrationBase.java [hunk 1]
```diff
@@ -189,6 +190,15 @@ String[] properties() default {
...
   protected abstract StorageConfigInfo getStorageConfigInfo();
 
+  /**
+   * A hook for test sub-classes to initialize {@link FileIO} objects used by test cases thenselves
+   * for accessing test storage.
+   */
+  protected void initializeClientFileIO(FileIO fileIO) {
+    fileIO.initialize(Map.of());
+  }
+  ;
```
Comment (eric-maynard) [inline integration-tests/src/main/java/org/apache/polaris/service/it/test/PolarisRestCatalogIntegrationBase.java]: ;)

```suggestion

```

File: runtime/service/src/intTest/java/org/apache/polaris/service/quarkus/it/QuarkusRestCatalogMinIoSpecialIT.java [hunk 2]
```diff
@@ -78,10 +78,10 @@
 import software.amazon.awssdk.services.s3.model.GetObjectResponse;
 
 @QuarkusIntegrationTest
-@TestProfile(QuarkusRestCatalogMinIoIT.Profile.class)
+@TestProfile(QuarkusRestCatalogMinIoSpecialIT.Profile.class)
 @ExtendWith(MinioExtension.class)
 @ExtendWith(PolarisIntegrationTestExtension.class)
-public class QuarkusRestCatalogMinIoIT {
+public class QuarkusRestCatalogMinIoSpecialIT {
```
Comment (eric-maynard) [inline runtime/service/src/intTest/java/org/apache/polaris/service/quarkus/it/QuarkusRestCatalogMinIoSpecialIT.java]: Might be worth a javadoc, it was not immediately obvious to me why this is _Special_

File: integration-tests/src/main/java/org/apache/polaris/service/it/test/PolarisRestCatalogIntegrationBase.java [hunk 3]
```diff
@@ -189,6 +190,15 @@ String[] properties() default {
    */
   protected abstract StorageConfigInfo getStorageConfigInfo();
 
+  /**
+   * A hook for test sub-classes to initialize {@link FileIO} objects used by test cases thenselves
+   * for accessing test storage.
+   */
+  protected void initializeClientFileIO(FileIO fileIO) {
```
Comment (eric-maynard) [inline integration-tests/src/main/java/org/apache/polaris/service/it/test/PolarisRestCatalogIntegrationBase.java:197]: nit: this PR uses both `Io` and `IO`

**Review by eric-maynard — DISMISSED** [review]

File: integration-tests/src/main/java/org/apache/polaris/service/it/test/PolarisRestCatalogIntegrationBase.java (hunk 1 above)
Comment (dimas-b) [reply integration-tests/src/main/java/org/apache/polaris/service/it/test/PolarisRestCatalogIntegrationBase.java]: oops - fixed

File: runtime/service/src/intTest/java/org/apache/polaris/service/quarkus/it/QuarkusRestCatalogMinIoSpecialIT.java (hunk 2 above)
Comment (dimas-b) [reply runtime/service/src/intTest/java/org/apache/polaris/service/quarkus/it/QuarkusRestCatalogMinIoSpecialIT.java]: javadoc added

File: integration-tests/src/main/java/org/apache/polaris/service/it/test/PolarisRestCatalogIntegrationBase.java (hunk 3 above)
Comment (dimas-b) [reply integration-tests/src/main/java/org/apache/polaris/service/it/test/PolarisRestCatalogIntegrationBase.java:197]: changed to `MinIO`

**Review by snazy — APPROVED** [review]
